Cargo.toml

```diff
@@ -16,5 +16,5 @@ version = "0.9.2"
 features = ["parallel"]
 
 [dependencies.pyo3]
-version = "0.20.0"
-features = ["extension-module"]
+version = "0.22.2"
+features = ["extension-module", "py-clone"]
```

Comment:
Setting the `py-clone` feature flag here (following https://pyo3.rs/v0.22.2/migration#pyclone-is-now-gated-behind-the-py-clone-feature) to avoid these errors in pyo3 0.22:

```rust
error[E0277]: the trait bound `Py<PyAny>: Clone` is not satisfied
  --> src/adapters.rs:35:5
   |
33 | #[derive(Clone)]
   |          ----- in this derive macro expansion
34 | pub(crate) struct PyFileObject {
35 |     file_obj: pyo3::PyObject,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `Py<PyAny>`
   |
   = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Py<PyAny>: Clone` is not satisfied
  --> src/adapters.rs:36:5
   |
33 | #[derive(Clone)]
   |          ----- in this derive macro expansion
...
36 |     write_fn: Option<pyo3::PyObject>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `Py<PyAny>`, which is required by `Option<Py<PyAny>>: Clone`
   |
   = note: required for `Option<Py<PyAny>>` to implement `Clone`
   = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Py<PyAny>: Clone` is not satisfied
  --> src/adapters.rs:37:5
   |
33 | #[derive(Clone)]
   |          ----- in this derive macro expansion
...
37 |     read_fn: Option<pyo3::PyObject>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `Py<PyAny>`, which is required by `Option<Py<PyAny>>: Clone`
   |
   = note: required for `Option<Py<PyAny>>` to implement `Clone`
   = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Py<PyAny>: Clone` is not satisfied
  --> src/adapters.rs:38:5
   |
33 | #[derive(Clone)]
   |          ----- in this derive macro expansion
...
38 |     readinto_fn: Option<pyo3::PyObject>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `Py<PyAny>`, which is required by `Option<Py<PyAny>>: Clone`
   |
   = note: required for `Option<Py<PyAny>>` to implement `Clone`
   = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `lazrs` (lib) due to 4 previous errors
```